pkg/cmd/attestation/verification/sigstore.go

```diff
@@ -65,13 +68,57 @@ func (v *LiveSigstoreVerifier) chooseVerifier(b *bundle.ProtobufBundle) (*verify
 	}
 	issuer := leafCert.Issuer.Organization[0]
 
-	// if user provided a custom trusted root file path, use the custom verifier
-	if v.config.CustomTrustedRoot != "" {
-		customVerifier, err := newCustomVerifier(v.config.CustomTrustedRoot)
+	if v.config.TrustedRoot != "" {
+		customTrustRoots, err := os.ReadFile(v.config.TrustedRoot)
 		if err != nil {
-			return nil, "", fmt.Errorf("failed to create custom verifier: %v", err)
+			return nil, "", fmt.Errorf("unable to read file %s: %v", v.config.TrustedRoot, err)
+		}
+
+		scanner := bufio.NewScanner(bytes.NewReader(customTrustRoots))
+		for scanner.Scan() {
+			// Load each trusted root
+			trustedRoot, err := root.NewTrustedRootFromJSON(scanner.Bytes())
+			if err != nil {
+				return nil, "", fmt.Errorf("failed to create custom verifier: %v", err)
+			}
+
+			// Compare bundle leafCert issuer with trusted root cert authority
+			certAuthorities := trustedRoot.FulcioCertificateAuthorities()
+			for _, certAuthority := range certAuthorities {
```

Comment:
Trusted roots are different from bundles in that they have a list of certificate authorities. The way this works for the Sigstore public good instance and the GitHub instance today is that the entries are for the same entity with slightly overlapping `validFor` configuration (to support rotating key material).

So if one of the certificate authority entries root cert has an issuer that matches, we guess that the trusted root that includes those certificate authority entries is the one we want.

Reply:
While this works for now, I think a slightly more robust method (that also works for Proxima) is to compare the subscriber cert's `AuthorityKeyId` with the "lowest" intermediate's `SubjectKeyId` (i.e the intermediate from the CA in the trust root). After we get a hit, we can then look at the issuer to understand GitHub vs PGI. 

Each stamp in Proxima will share the CA and first intermediate certificate with dotcom.

Reply:
Yeah, and this is further complicated by different trusted roots having different chains! I added a `getLowestCertInChain()` function to support future use-cases, without breaking on a custom trusted root that only has a root certificate.